archive/archive_ingest/src/archive_ingest.py

```diff
@@ -0,0 +1,68 @@
+# -*- encoding: utf-8 -*-
+"""
+Receives a message to ingest a bag giving the URL and publishes the archive event to an SNS topic.
+"""
+
+import os
+
+import boto3
+import daiquiri
+from urllib.parse import urlparse
+
+from wellcome_aws_utils.sns_utils import publish_sns_message
+from wellcome_aws_utils.lambda_utils import log_on_error
+
+daiquiri.setup(level=os.environ.get('LOG_LEVEL', 'INFO'))
+logger = daiquiri.getLogger()
+
+
+def ensure_valid_request(event):
+    required_key = 'bagURL'
+    if type(event) is not dict:
+        raise ValueError(f"Invalid request not json dict: {event}")
+    if required_key not in event:
+        raise ValueError(f"Invalid request missing {required_key}")
```

Comment:
Python idiom is Easier to Ask For Permission, not Look Before You Leap.

It would be better to assume this is an object that implements square-bracket lookup (the `__getitem__` method), and has a key called `bagURL`, and catch the error, if any.

i.e.

```python
try:
    bag_url = event['bagURL']
except KeyError:
    raise ValueError('custom error goes here')
```

although KeyError is probably the most correct error to raise here.

Other notes:

* In general, prefer `isinstance(obj, expected_type)` to `type(obj) is expected_type` – the former allows for subclassing, the latter doesn't
* If the required key changes in `main()`, this function could pass if you forget to update `required_key`. Better to pass it in as a parameter, or catch the error upon lookup.